### PR TITLE
Implement 'duration' field into Curriculum Catalog

### PR DESCRIPTION
**Pull request by TurnerRiley**
Calculates and shows the `duration` field for each Curriculum Catalog card (previously, each value just defaulted to `school_year`).

Each unit sums up the length of its lessons (in minutes), then the course offering sums up the length of the units (in the case of a UnitGroup), and then it is converted from a number of minutes to a label based on pre-defined bands:
- Lesson: 0 - 90 minutes
- Week: 91 - 250 minutes
- Month: 251 - 950 minutes
- Quarter: 951 - 2500 minutes
- Semester: 2501 - 5000 minutes
- School year 5001+ minutes (since there's not an explicit cap on what defines a school year course, I just set the cap to be the number of minutes in a year)

### Previously with hard-coded `school_year`
old_duration

### Now with dynamic defining of `duration`
New_duration

## Links
Spec: [here](https://docs.google.com/document/d/1UID-xzCF5gxOHihcYiYuZwpotCpGXGPTzqX1UyS37-I/edit#bookmark=kix.wbpfip3hhboe)
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?modal=detail&selectedIssue=ACQ-515&assignee=60d62161f65054006980bd71)

## Testing story
Local testing and adding unit tests.

## PR Checklist:
- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by dmcavoy** [comment]
@TurnerRiley which courses did not have a duration and are being hidden?

**Comment by TurnerRiley** [comment]
> @TurnerRiley which courses did not have a duration and are being hidden?

Oh I just removed the `!!curriculum.duration` filtering out. I previously had it because I thought there were some without a duration that were causing an issue with rendering the page, but that's no longer the case.

File: dashboard/test/models/course_offering_test.rb [hunk 1]
```diff
@@ -571,6 +571,52 @@ class CourseOfferingTest < ActiveSupport::TestCase
     Unit.clear_cache
   end
 
+  test 'duration returns label associated with sum of units duration' do
+    # Create a unit with multiple lessons, each with a different number of lesson activities.
+    unit = create(:script, name: 'unit1', family_name: 'test-duration', version_year: '1997', is_course: true, published_state: 'stable')
+    lesson_group = create(:lesson_group, key: 'lesson_group1', script: unit)
+
+    lesson1 = create(:lesson, script: unit, lesson_group: lesson_group)
```
Comment (davidsbailey) [inline dashboard/test/models/course_offering_test.rb:579]: you can simplify some of this:
```
unit = create :script, :with_lessons, ...
create(:lesson_activity, lesson: unit.lessons.first, duration: 40)
```

**Review by davidsbailey — APPROVED** [review]
nice work Turner!

**Pull request merged**